jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java

```diff
@@ -670,12 +668,25 @@ private String getIdentity() {
         }
     }
 
-    public void createGlobalDirOnNewProject(@Observes NewProjectEvent newProjectEvent) {
+    private String makeGlobalDirPath(final String projectPath,
+                                     final String separator) {
+        if (projectPath == null || projectPath.trim().isEmpty()) {
+            return "global";
+        } else if (!projectPath.endsWith(separator)) {
+            return projectPath + separator + "global";
+        } else {
+            return projectPath + "global";
+        }
+    }
+
+    public void createGlobalDirOnNewProject(@Observes final NewModuleEvent newModuleEvent) {
         // create the global dir before the asset is created (upon new project creation)
-        KieProject project = (KieProject) newProjectEvent.getProject();
-        String projectPath = org.uberfire.backend.server.util.Paths.convert(project.getRootPath()).toUri().toString().replaceAll("\\s", "%20");
-        String separator = org.uberfire.backend.server.util.Paths.convert(project.getRootPath()).getFileSystem().getSeparator();
-        String globalDirPath = projectPath + separator + "global";
+        KieModule module = (KieModule) newModuleEvent.getModule();
+        String projectPath = org.uberfire.backend.server.util.Paths.convert(module.getRootPath()).toUri().toString().replaceAll("\\s", "%20");
```

Comment:
@tsurdilo I've to merge this conflict with https://github.com/kiegroup/jbpm-designer/commit/b27e821bdaee2571bc541925da9172d4b92ce78c 

Could you please take a look on  this?